tests/mock_ssh.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+
+from socket import AF_INET, IPPROTO_TCP, SO_REUSEADDR, SOCK_STREAM, SOL_SOCKET, socket
+
+from threading import Event, RLock
+
+from paramiko import (
+    AUTH_FAILED,
+    AUTH_SUCCESSFUL,
+    OPEN_FAILED_ADMINISTRATIVELY_PROHIBITED,
+    OPEN_SUCCEEDED,
+    AutoAddPolicy,
+    RSAKey,
+    ServerInterface,
+    SSHClient,
+    Transport,
+)
+
+class MockSSHServer(ServerInterface):
+    """An ssh server accepting the pre-generated key."""
+
+    ssh_username = "pytest"
+    ssh_key = RSAKey.generate(4096)
+
+    def __init__(self):
+        self._channel = None
+        self._client = None
+        self._command = None
+        self.event = Event()
+        self._server_transport = None
+        self._thread_lock = RLock()
+
+    def __del__(self):
+        self.close()
+
+    @property
+    def client(self):
+        with self._thread_lock:
+            if self._client is None:
+                connection_event = Event()
+
+                server_socket = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP)
+                server_socket.setsockopt(SOL_SOCKET, SO_REUSEADDR, 1)
+                server_socket.bind(("127.0.0.1", 0))
+                server_socket.listen()
+
+                server_address = server_socket.getsockname()
+
+                client_socket = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP)
+                client_socket.connect(server_address)
+
+                (transport_socket, _) = server_socket.accept()
+
+                self._server_transport = Transport(transport_socket)
+                self._server_transport.add_server_key(self.__class__.ssh_key)
+                self._server_transport.start_server(connection_event, self)
+
+                self._client = SSHClient()
+                self._client.set_missing_host_key_policy(AutoAddPolicy())
+
+                self._client.connect(
+                    server_address[0],
+                    server_address[1],
+                    username=self.__class__.ssh_username,
+                    pkey=self.__class__.ssh_key,
+                    sock=client_socket
+                )
+
+                connection_event.wait()
+
+        return self._client
+
+    def check_channel_request(self, kind, chanid):
+        if kind == "session":
+            return OPEN_SUCCEEDED
+        return OPEN_FAILED_ADMINISTRATIVELY_PROHIBITED
+
+    def check_auth_password(self, username, password):
+        return AUTH_FAILED
+
+    def check_auth_publickey(self, username, key):
+        return (
+            AUTH_SUCCESSFUL
+            if username == self.__class__.ssh_username and key == self.__class__.ssh_key else
+            AUTH_FAILED
+        )
```

Comment:
Why not
```
if username == self.__class__.ssh_username and key == self.__class__.ssh_key:
    return AUTH_SUCCESSFUL
return AUTH_FAILED
```
Ternary operator is not useful here and only makes it harder to read

Reply:
I personally do not agree to your opinion about the ternary operator for that example but do agree based on the enforced formatting that your code is more readable in that case :)